### PR TITLE
Update react transition group

### PR DESCRIPTION
**Pull request by nLight**

### CODE REVIEW

**Comment by TattdCodeMonkey** [comment]
This looks good, but should it be a `feat` commit or a `BREAKING CHANGE`?

**Comment by nLight** [comment]
@TattdCodeMonkey it's both:
```
feat: update react-transition-group@2.5.0

BREAKING CHANGE: transitionLeaveTimeout -> transitionExitTimeout
```

The only problem with the change is that _all_ transitions are gone... so something is missing. either css or js setup

File: src/Dropdown/Dropdown.js [hunk 1]
```diff
@@ -395,23 +396,26 @@ class Dropdown extends Util.mixin(BindMixin) {
...
     }
 
-    if (state.renderHidden) {
-      dropdownMenu = (
-        <div key="concealer" className="dropdown-menu-concealer">
-          {dropdownMenu}
-        </div>
-      );
-    }
+    // if (state.renderHidden) {
```
Comment (nLight) [inline src/Dropdown/Dropdown.js]: @TattdCodeMonkey this is used to calc height of the list and then to decided whether it should open up or down. It's broken

File: src/Dropdown/Dropdown.js [hunk 2]
```diff
@@ -395,23 +396,26 @@ class Dropdown extends Util.mixin(BindMixin) {
...
+    // }
 
     if (props.transition) {
       dropdownMenu = (
-        <CSSTransitionGroup
-          transitionName={transitionName}
-          transitionEnterTimeout={props.transitionEnterTimeout}
-          transitionLeaveTimeout={props.transitionLeaveTimeout}
+        <CSSTransition
+          in={state.isOpen && !state.renderHidden}
```
Comment (nLight) [inline src/Dropdown/Dropdown.js]: That was the important bit. I guess we can move `!state.renderHidden` to the line 407 otherwise it transitions while being hidden

File: src/Modal/ModalContents.js [hunk 3]
```diff
@@ -380,18 +381,20 @@ class ModalContents extends Util.mixin(BindMixin) {
...
-        transitionAppear={props.transitionAppear}
-        transitionEnter={props.transitionEnter}
-        transitionLeave={props.transitionLeave}
-        transitionName={props.transitionNameModal}
-        transitionAppearTimeout={props.transitionAppearTimeoutModal}
-        transitionEnterTimeout={props.transitionEnterTimeoutModal}
-        transitionLeaveTimeout={props.transitionLeaveTimeoutModal}
-        component="div"
+      <CSSTransition
+        in={props.open}
```
Comment (nLight) [inline src/Modal/ModalContents.js]: modal should be alrighg

**Comment by nLight** [comment]
@TattdCodeMonkey 

TLDR couldn't figure out `List` `ListItem` though
dropdown transitions but fails to detect if it should be opening `up`
modal works
this is how it should be http://mesosphere.github.io/reactjs-components/

**Comment by nLight** [comment]
testing `npm run livereload`

File: src/Dropdown/Dropdown.js (hunk 1 above)
Comment (TattdCodeMonkey) [reply src/Dropdown/Dropdown.js]: Figured this out. It is because the ref's were broken and not updated for React 16.

Comment (nLight) [reply src/Dropdown/Dropdown.js]: oh well 😅

**Comment by mperrotti** [comment]
About the `List`/`ListItem` transition in:
I see we updated the styles for a list item animating out, but I don't see any changes for animating in.

However, I think the slide in animation is slow and unnecessary, and has a negative impact on perceived performance. I recommend we remove the ListItem transition.

**Comment by mperrotti** [comment]
Getting a few console errors from props being passed from form components to their dom nodes. A good example is `textarea`.
To be fair, this was happening before these changes, so I won't let that block this particular PR.
